File: app/src/main/java/com/adiaz/deportesmadrid/adapters/GroupsAdapter.java
package com.adiaz.deportesmadrid.adapters;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.adiaz.deportesmadrid.R;
import com.adiaz.deportesmadrid.db.entities.Competition;
import com.adiaz.deportesmadrid.utils.UtilsPreferences;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class GroupsAdapter extends RecyclerView.Adapter<GroupsAdapter.ViewHolder> {

    Context mContext;
    List<Competition> mCompetitionList;
    private final GroupsAdapter.ListItemClickListener mOnClickListener;


    public GroupsAdapter(Context mContext, List<Competition> competitionList, ListItemClickListener mOnClickListener) {
        this.mContext = mContext;
        this.mCompetitionList = competitionList;
        this.mOnClickListener = mOnClickListener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        View view = layoutInflater.inflate(R.layout.listitem_group, parent, false);
        return new GroupsAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.tvGroupName.setText(mCompetitionList.get(position).nomGrupo());
        holder.tvCompetition.setText(mCompetitionList.get(position).nomCompeticion());
        holder.tvFase.setText(mCompetitionList.get(position).nomFase());
        if (UtilsPreferences.isShowCompetitionsNumber(mContext)) {
            holder.tvGroupName.append(" (" + mCompetitionList.get(position).id() + ")");
        }
    }

    @Override
    public int getItemCount() {
        return mCompetitionList.size();
    }

    public interface ListItemClickListener {
        void onListItemClick(int clickedItemIndex);
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        @BindView(R.id.tv_listitem_title)
        TextView tvGroupName;

        @BindView(R.id.cv_listitem)
        CardView cvListItem;

        @BindView(R.id.tv_competition)
        TextView tvCompetition;

        @BindView(R.id.tv_fase)
        TextView tvFase;

        /*        @BindView(R.id.tv_sport_count)
        TextView tvCompetitionCount;*/

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            cvListItem.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            int position = getAdapterPosition();
            mOnClickListener.onListItemClick(position);
        }
    }

}

File: app/src/main/java/com/adiaz/ligasmadrid/adapters/GroupsAdapter.kt
package com.adiaz.ligasmadrid.adapters

import android.content.Context
import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.adiaz.ligasmadrid.R
import com.adiaz.ligasmadrid.db.entities.Group
import com.adiaz.ligasmadrid.extensions.layoutInflater
import com.adiaz.ligasmadrid.utils.UtilsPreferences

import kotlinx.android.synthetic.main.listitem_group.view.*

class GroupsAdapter(var mContext: Context, var mGroupList: List<Group>?, mOnClickListener: ListItemClickListener) :
        RecyclerView.Adapter<GroupsAdapter.ViewHolder>() {

    private val mOnClickListener: GroupsAdapter.ListItemClickListener = mOnClickListener

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = mContext.layoutInflater.inflate(R.layout.listitem_group, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvGroupName.text = mGroupList!![position].nomGrupo
        holder.tvCompetition.text = mGroupList!![position].nomCompeticion
        holder.tvFase.text = mGroupList!![position].nomFase
        if (UtilsPreferences.isShowCompetitionsNumber(mContext)) {
            holder.tvGroupName.append(" (" + mGroupList!![position].id + ")")
        }
    }

    override fun getItemCount(): Int {
        return mGroupList!!.size
    }

    interface ListItemClickListener {
        fun onListItemClick(clickedItemIndex: Int)
    }

    inner class ViewHolder (itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        val tvGroupName: TextView = itemView.tvGroupName
        val cvListItem: CardView = itemView.cardView
        val tvCompetition: TextView = itemView.tvCompetition
        val tvFase: TextView = itemView.tvFase

        init {
            cvListItem.setOnClickListener(this)
        }
        override fun onClick(view: View) {
            val position = adapterPosition
            mOnClickListener.onListItemClick(position)
        }
    }

}
